class-two-factor-core.php

```diff
@@ -825,17 +825,29 @@ public static function login_html( $user, $login_nonce, $redirect_to, $error_msg
 					<?php esc_html_e( 'Having Problems?', 'two-factor' ); ?>
 				</p>
 				<ul>
-					<?php
-					foreach ( $backup_providers as $backup_provider_key => $backup_provider ) :
+				<?php
+					$links = [];
+
+					foreach ( $backup_providers as $backup_provider_key => $backup_provider ) {
 						$backup_link_args['provider'] = $backup_provider_key;
-						?>
-						<li>
-							<a href="<?php echo esc_url( self::login_url( $backup_link_args ) ); ?>">
-								<?php echo esc_html( $backup_provider->get_alternative_provider_label() ); ?>
-							</a>
-						</li>
-					<?php endforeach; ?>
-				</ul>
+						$links[] = sprintf(
+							'<li><a href="$1%s">$2%s</a></li>',
+							esc_url( self::login_url( $backup_link_args ) ),
+							esc_html( $backup_provider->get_alternative_provider_label() )
+						);
+					}
+
+					/*
+					* Allow plugins to add links to the two-factor login form.
+					*/
+					$links = apply_filters( 'two_factor_login_support_links', $links );
+
+					// Echo out the filtered links
+					foreach ( $links as $link ) {
+						echo wp_kses_post( $link );
+					}
```

Comment:
```suggestion
					/*
					* Allow plugins to filter the backup method links on the login form.
					*/
					$links = apply_filters( 'two_factor_login_backup_links', $links );

					// Echo out the filtered links
					echo implode( '', $links );
```

Given this filter only runs when there are backup providers, calling it a support links might be confusing for some, we can probably name it something like `two_factor_login_backup_links`?

There's probably an argument that this should be built prior to the `if ( $backup_providers )` conditional to output filtered data even when there exist no backup methods for the given user?

I don't think we need to explicitly sanitize the HTML here either, it's either hard-coded HTML or from a PHP function that can do anything anyway.

Reply:
These points remain:

> Given this filter only runs when there are backup providers, calling it a support links might be confusing for some, we can probably name it something like two_factor_login_backup_links?

OR 

> There's probably an argument that this should be built prior to the if ( $backup_providers ) conditional to output filtered data even when there exist no backup methods for the given user?

I suspect the intention of the filter was actually to be used when the user has no backup provider, or, at least that others who would have a use for it would be in that group.